<?xml version="1.0" ?>
<launch>
  <!-- params -->
  <param name="forest_file" value="$(find leg_tracker)/config/trained_leg_detector_RForest_sim.yaml" />
  <param name="scan_frequency" value="10"/>  
  <param name="use_sim_time" value="false" />

  <param name="scan_topic" value="/scan_multi" />
  <param name="fixed_frame" value="odom"/>

  <arg name="detection_threshold" value="0.50"/>
  <arg name="max_detect_distance" value="29" />  
  <arg name="use_scan_header_stamp_for_tfs" value="false"/>

  <!-- run detect_leg_clusters -->  
  <node pkg="leg_tracker" type="detect_leg_clusters" name="detect_leg_clusters" output="screen">
    <param name="min_points_per_cluster" value="3" />
    <param name="cluster_dist_euclid" value="0.13" />
    <param name="detection_threshold" value="$(arg detection_threshold)"/>
    <param name="use_scan_header_stamp_for_tfs" value="$(arg use_scan_header_stamp_for_tfs)"/>
    <param name="max_detect_distance" value="$(arg max_detect_distance)"/>
  </node>

  <!-- Merge laser scans -->
  <node pkg="ira_laser_tools" name="laserscan_multi_merger" type="laserscan_multi_merger" output="screen">
    <param name="destination_frame" value="base_link"/>
    <param name="cloud_destination_topic" value="/merged_cloud"/>
    <param name="scan_destination_topic" value="/scan_multi"/>
    <param name="laserscan_topics" value ="/f_scan /b_scan" /> <!-- LIST OF THE LASER SCAN TOPICS TO SUBSCRIBE -->
    <param name="angle_min" value="-3.14"/>
    <param name="angle_max" value="3.14"/>
    <param name="angle_increment" value="0.0058"/>
    <param name="scan_time" value="0.0333333"/>
    <param name="range_min" value="0.30"/>
    <param name="range_max" value="50.0"/>
  </node>

  <!-- run joint_leg_tracker -->
  <node pkg="leg_tracker" type="joint_leg_tracker.py" name="joint_leg_tracker" output="screen">
    <remap from="/local_map" to="/map" />
  </node>
  
  <!-- mapping -->
  <node pkg="iris_lama_ros" type="slam2d_ros" name="slam2d_ros" output="screen">
    <param name="scan_topic" value="/proc_scan_multi"/>
    <param name="mrange" value="0"/>
    <remap from="/local_map" to="/map" />
  </node>

  <!-- run rviz -->                
  <!-- <node name="rviz" pkg="rviz" type="rviz" args="-d $(find leg_tracker)/config/visualize.rviz" required="false"/> -->
  
</launch>
